bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/ThingManager.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.thing;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+
+/**
+ * {@link ThingManager} interface defines a method for disabling or enabling a {@link Thing}.
+ * 
+ * @author Yordan Zhelev - Initial contribution
+ */
+public interface ThingManager {
+
+    /**
+     * If {@code disabled} parameter is set to {@code true} the status of the given {@link Thing} will be updated to
+     * {@code UNINITIALIZED.DISABLED}.
+     * If {@code disabled} parameter is set to {@code false} {@link ThingHandler#initialize()} will be invoked for the
+     * given {@link Thing}. Enabling is implemented as a non-blocking operation and the method will return without
+     * waiting for the {@link Thing}'s status to be updated.
+     * 
+     * @param thing the {@link Thing} to be disabled or enabled.
+     * @param disabled if set to {@code true} the corresponding thing will be disabled. If set to {@code false} the
+     *            thing will be enabled.
+     */
+    void setDisabled(final @NonNull Thing thing, final boolean disabled);
```

Comment:
If we decide that when the thing is NOT with status detail DISABLED, then, it can be considered as *enabled*, then I could rename the method to not have *disabled* in its name. Maybe just *enableThing()*.